packages/expo-camera/android/src/main/java/expo/modules/camera/CameraViewHelper.kt

```diff
@@ -108,6 +108,29 @@ object CameraViewHelper {
     return exifMap
   }
 
+  @JvmStatic
+  fun setExifData(baseExif: ExifInterface, exifMap: Map<String, Any>) {
+    for ((_, name) in exifTags) {
+      exifMap[name]?.let {
+        // Convert possible type to string before putting into baseExif
+        when(it::class.simpleName) {
+          "String" -> baseExif.setAttribute(name, it as String)
+          "Double" -> baseExif.setAttribute(name, (it as Double).toBigDecimal().toPlainString())
+          "Int" -> baseExif.setAttribute(name, (it as Int).toBigDecimal().toPlainString())
+          "Boolean" -> baseExif.setAttribute(name, (it as Boolean).toString())
+        }
```

Comment:
Kotlin can do the type-check and inference for you 😉 
Also, both `Int` and `Double` inherit from `Number` so you can kill two birds with one stone
```suggestion
        when(it) {
          is String -> baseExif.setAttribute(name, it)
          is Number -> baseExif.setAttribute(name, it.toBigDecimal().toPlainString())
          is Boolean -> baseExif.setAttribute(name, it.toString())
        }
```

Reply:
Thanks @barthap , updated.